src/components/thumbnail-card/ThumbnailCard.js

```diff
@@ -1,23 +1,35 @@
 // @flow
 import * as React from 'react';
+import classNames from 'classnames';
 
 import ThumbnailCardDetails from './ThumbnailCardDetails';
 import ThumbnailCardThumbnail from './ThumbnailCardThumbnail';
 
 import './ThumbnailCard.scss';
 
 type Props = {
+    actionItem?: React.Node,
     className?: string,
     icon?: React.Node,
+    shouldExpandOnHover?: boolean,
     subtitle?: React.Node,
     thumbnail: React.Node,
     title: React.Node,
 };
 
-const ThumbnailCard = ({ className = '', icon, subtitle, title, thumbnail, ...rest }: Props) => (
-    <div className={`thumbnail-card ${className}`} {...rest}>
+const ThumbnailCard = ({
+    actionItem,
+    className = '',
+    icon,
+    shouldExpandOnHover = false,
+    subtitle,
+    title,
+    thumbnail,
+    ...rest
+}: Props) => (
+    <div className={classNames('thumbnail-card', className, { 'is-expandable': shouldExpandOnHover })} {...rest}>
         <ThumbnailCardThumbnail thumbnail={thumbnail} />
-        <ThumbnailCardDetails icon={icon} subtitle={subtitle} title={title} />
+        <ThumbnailCardDetails actionItem={actionItem} icon={icon} subtitle={subtitle} title={title} />
```

Comment:
Thoughts on following the pattern `Media` has with `MediaBody`, `MediaMenu` instead of passing React nodes down as props?

Reply:
Using named sub-components allows us to compartmentalize changes, and give users some guardrails for making cleaner implementations. Also gives us an enforcement mechanism that we can `throw` against